gordo_components/client/client.py

```diff
@@ -249,33 +224,47 @@ def download_model(self) -> typing.Dict[str, BaseEstimator]:
         """
         models = dict()
         for endpoint in self.endpoints:
-            resp = self.session.get(f"{endpoint.endpoint}/download-model")
+            resp = self.session.get(
+                f"{self.base_url + endpoint.endpoint}/download-model"
+            )
             if resp.ok:
-                models[endpoint.target_name] = serializer.loads(resp.content)
+                models[endpoint.endpoint_metadata.metadata.name] = serializer.loads(
+                    resp.content
+                )
             else:
                 raise IOError(f"Failed to download model: '{repr(resp.content)}'")
         return models
 
-    def get_metadata(self) -> typing.Dict[str, dict]:
+    def get_metadata(self, force_refresh: bool = False) -> typing.Dict[str, dict]:
         """
         Get the metadata for each target
 
+        Parameters
+        ----------
+        force_refresh : bool
+            Even if the previous request was cached, make a new request for metadata.
+
         Returns
         -------
         Dict[str, dict]
             Mapping of target names to their metadata
         """
-        metadata = dict()
-        for endpoint in self.endpoints:
-            resp = self.session.get(f"{endpoint.endpoint}/metadata")
-            if resp.ok:
-                metadata[endpoint.target_name] = resp.json()
-            else:
-                raise IOError(f"Failed to get metadata: '{repr(resp.content)}'")
-        return metadata
+        if hasattr(self, "_metadata") and not force_refresh:
+            return self._metadata.copy()
```

Comment:
Rather than using the `_` prefix, we could maybe postfix it, as sklearn does when certain variables are only available after specific methods are called. ie. `fit()`. I lean every so slightly to keeping it in here, rather than 'polluting' the class with variables people don't really need to be concerned with outside of this scope. :man_shrugging: